### DISCUSSION TITLE
Create an RSS feed with content:encoded

### DISCUSSION THREAD

**Discussion by ryadbkr** [alexdebril/feed-io#369] [Q&A] [answered]
We've been trying to create an RSS feed containing 
`<content:encoded><![CDATA[ <p> no luck ]]></content:encoded>` 

we've tried creating an element with the CDATA as content but it seems the CDATA tag is being encoded `(<content:encoded>&lt;![CDATA[),` hence it's not well interpreted by the RSS/XML parser,

Is using **CDATA** within `<content:encoded>` supported? If no are you planing on adding it.

Thanks

**Comment by alexdebril** [comment, marked as answer]
Hi,

No, it's not supported and I don't plan to do so. If you need that feature you can:
- build your own standard and add it in the Specification provided to FeedIo
- submit a Pull Request to make feed-io support it in its actual standards

But I'm not in favor of the second one. It's an edge-case, I prefer to avoid having to support it in the long run.

**Reply by ryadbkr** [reply]
Ok, for futur notes, 
I added my own rule to transform certain elements to CDATA, but it was not sufficient since te default XML formatter https://github.com/alexdebril/feed-io/blob/main/src/FeedIo/Formatter/XmlFormatter.php#L65
always overrides it with an optionalField rule, so I had to also create a custom formatter in order to fix that.


```
        $rules = [];
        $optionalFields = $node->listElements();
        foreach ($optionalFields as $optionalField) {
            $rules[$optionalField] = new OptionalField($optionalField);
        }

        return array_merge($rules, $ruleSet->getRules());
```

**Comment by fritzmg** [comment]
Btw. one use case would be [Facebook Instant Articles](https://developers.facebook.com/docs/instant-articles/publishing/setup-rss-feed) for which the content of the article needs to be in a `<content:encoded>` node with `<![CDATA[ ... ]]>` (at least according to their docs).

**Reply by alexdebril** [reply]
Sure thing, if it's really something valuable for your project then I think it's worth submitting a pull request 🙂
